mongorestore/mongorestore_test.go

```diff
@@ -250,6 +251,110 @@ func TestMongorestore(t *testing.T) {
 	})
 }
 
+func TestMongoRestoreSpecialCharactersNameSpaces(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("With a test MongoRestore", t, func() {
+		args := []string{
+			NumParallelCollectionsOption, "1",
+			NumInsertionWorkersOption, "1",
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		db := session.Database("db1")
+		Convey("and majority is used as the default write concern", func() {
+			So(db.WriteConcern(), ShouldResemble, writeconcern.New(writeconcern.WMajority()))
+		})
+
+		longCollection := db.Collection(specialCharactersCollectionName)
```

Comment:
```suggestion
		specialCollection := db.Collection(specialCharactersCollectionName)
```
[nit] copy-paste naming